src/server/console/default_configuration.cpp

```diff
@@ -102,6 +103,7 @@ std::shared_ptr<mir::ConsoleServices> mir::DefaultServerConfiguration::the_conso
                     std::make_unique<RealVTFileOperations>(),
                     std::make_unique<RealPosixProcessOperations>(),
                     the_options()->get<int>(options::vt_option_name),
+                    *the_emergency_cleanup(),
```

Comment:
Yeah. I think there are a couple of places where we hold a `shared_ptr` in an object merely to prevent it from being destroyed and recreated, even though we only use it in the constructor.

I wonder if switching `CachedPtr` to storing a `shared_ptr` would break anything…

Reply:
PR #392 suggests that `ServerConfiguration` having shared ownership of the configuration does not, in fact, break anything.

Reply:
At the risk of continuing a digression I started about repeating a pre-existing issue...

The problem is we've subverted the the Configuration::the_foo() mechanism. It should be used for supplying objects that are needed *in the implementation* (and the shared_ptr<> is always stored in a data member). The intended purpose of the "Configuration" is to ensure that the same instance is supplied everywhere.

The usage here (and in a few other places) is needed *in the initialization*. This should be accessing an object owned, not by the "Configuration" but by the "Server". That is, in practice, what is happening but here it isn't expressed locally in the code.

**NB We don't need to address this to land the PR**